src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/VsManagedLanguagesProjectSystemServices.cs

```diff
@@ -194,6 +194,8 @@ private static LibraryDependency ToPackageLibraryDependency(PackageReference ref
         {
             var dependency = new LibraryDependency
             {
+                AutoReferenced = MSBuildStringUtility.IsTrue(GetReferenceMetadataValue(reference, ProjectItemProperties.IsImplicitlyDefined)),
+                GeneratePathProperty = MSBuildStringUtility.IsTrue(GetReferenceMetadataValue(reference, ProjectItemProperties.GeneratePathProperty)),
```

Comment:
You also need to update `_referenceMetadata` in order to populate `reference` instance with `GeneratePathProperty` metadata.

And please add tests to consume this property inside VS for PackageReference based projects (both SDK/ classic csproj), otherwise we might end up in broken state for commandLine vs VS.

Reply:
Done.  I don't see anywhere that your tests are verifying that PackageReference metadata is read correctly in Visual Studio.  Where should I add these tests?

I found [LegacyPackageReferenceProjectTests](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/ProjectSystems/LegacyPackageReferenceProjectTests.cs) but it doesn't seem related to `VsManagedLanguagesProjectSystemServices`

Is `VsManagedLanguagesProjectSystemServices` pulled into a test via MEF and I can't find it?

Reply:
You'll have to add End-to-end tests (Apex Tests) to cover this scenario inside Visual Studio. May be @nkolev92 can help you with that...

I'm also adding func tests for `LegacyPackageReferenceProject` which should add the relevant infrastructure to test these projects inside VS but that will also mock `VsManagedLanguagesProjectSystemServices` since it calls into Project System.